Firestore/Swift/Source/Codable/DocumentReference+ReadDecodable.swift

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2021 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+import FirebaseFirestore
+
+extension DocumentReference {
+  /// Encodes an instance of `Encodable` and overwrites the encoded data
+  /// to the document referred by this `DocumentReference`. If no document exists,
+  /// it is created. If a document already exists, it is overwritten.
+  ///
+  /// See `Firestore.Encoder` for more details about the encoding process.
+  ///
+  /// - Parameters:
+  ///   - value: An instance of `Encodable` to be encoded to a document.
+  ///   - encoder: An encoder instance to use to run the encoding.
+  ///   - completion: A block to execute once the document has been successfully
+  ///                 written to the server. This block will not be called while
+  ///                 the client is offline, though local changes will be visible
+  ///                 immediately.
+
+  public func getDocument<T: Decodable>(as type: T.Type,
```

Comment:
@ryanwilson - Should we consider making this our first async/await API?

Reply:
I can add that too. Only thing is that I don't know how to correctly annotate iOS 15 support on Xcode 13 and iOS 13+ support on Xcode 13.2. Does anyone have an example for this?

Reply:
I now added this too - but currently only for iOS 15 and friends.

Reply:
That's fine for now. We haven't found a good incantation that will also work for Xcode 13.0 because of its missing macOS support. We may end up requiring 13.1 or 13.2 for Swift libraries - and we can update all Swift Concurrency code globally, including this, when we make that decision.